libraries/AP_Common/tests/test_location.cpp

```diff
@@ -0,0 +1,309 @@
+#include <AP_gtest.h>
+#include <AP_Common/Location.h>
+#include <AP_Math/AP_Math.h>
+#include <AP_AHRS/AP_AHRS.h>
+#include <GCS_MAVLink/GCS_Dummy.h>
+#include <AP_BoardConfig/AP_BoardConfig.h>
+#include <AP_Logger/AP_Logger.h>
+#include <AP_ExternalAHRS/AP_ExternalAHRS.h>
+#include <AP_Vehicle/AP_Vehicle.h>
+#include <AP_Terrain/AP_Terrain.h>
+
+const AP_HAL::HAL& hal = AP_HAL::get_HAL();
+static GCS_Dummy _gcs;
+AP_Int32 logger_bitmask;
+static AP_Logger logger{logger_bitmask};
+
+const AP_Param::GroupInfo GCS_MAVLINK_Parameters::var_info[] = {
+        AP_GROUPEND
+};
+
+class DummyVehicle : public AP_Vehicle {
```

Comment:
Yowsers this is a lot of boiler-plate.  Why do we need a vehicle for this test suite?

Reply:
This is needed to be able to set/get home.
There is surely a better way to do that maybe with a mock class, but I am not used to that yet